Resources/Private/BackendModule/src/Sagas/AppSaga.ts

```diff
@@ -52,78 +64,168 @@ export function* watchStartModule() {
-    const language = yield select((state: StateInterface) => state.app.moduleConfiguration.language)
+function* generateItemSaga({ payload: item }: PayloadAction<StatefulModuleItem>) {
+    yield put(setItemGenerating({ identifier: item.identifier, generating: true }))
+    const language = yield getLanguageDependingOnScope(item)
     const externalService = ExternalService.getInstance()
     const translationService = TranslationService.getInstance()
-    try {
-        const response = yield externalService.generate('image_alt_text', language, [
-            {
-                identifier: 'url',
-                value: [
-                    item.fullsizeUri,
-                    item.thumbnailUri
+    const itemGenerationConfig = yield getItemGenerationConfigDependingOnScope(item)
+
+    yield* Object.keys(itemGenerationConfig).map(yield function* (propertyName: string) {
+        const sidekickConfiguration = itemGenerationConfig[propertyName]
+        try {
+            const propertyValue = yield externalService.generate(sidekickConfiguration.module, language, sidekickConfiguration.userInput)
+            console.log(propertyValue)
+            yield put(updateItemProperty({ identifier: item.identifier, propertyName, propertyValue }))
+        } catch (e) {
+            if (e instanceof AiAssistantError) {
+                yield put(setErrorMessage(translationService.translate('NEOSidekick.AiAssistant:Error:' + e.code, e.message, {0: e.externalMessage})))
+            }
+            // todo catch other errors with a more generic message?
+        }
+    })
+    yield put(setItemGenerating({ identifier: item.identifier, generating: false }))
+}
+
+function *autogenerateItemSaga(action: PayloadAction<StatefulModuleItem>) {
+    const { payload: item } = action
+    if (yield shouldGenerateAutomaticallyDependingOnScope(item)) {
+        yield generateItemSaga(action)
+    }
+}
+
+function *shouldGenerateAutomaticallyDependingOnScope(item: ModuleItem) {
+    const scope = yield select(state => state.app.scope)
+    const configuration: ModuleConfiguration = yield select(state => state.app.moduleConfiguration)
+    switch (scope) {
+        case 'altTextGeneratorModule': return true
+        case 'focusKeywordGeneratorModule':
+            return (
+                (item.focusKeyword === '' && configuration.generateEmptyFocusKeywords) ||
+                (item.focusKeyword !== '' && configuration.regenerateExistingFocusKeywords)
+            )
+    }
+}
+
+function *getLanguageDependingOnScope(item: ModuleItem) {
+    const scope = yield select(state => state.app.scope)
+    switch (scope) {
+        case 'altTextGeneratorModule': return yield select((state: StateInterface) => state.app.moduleConfiguration.language)
+        case 'focusKeywordGeneratorModule': return item.language
+    }
+}
+
+function *getItemGenerationConfigDependingOnScope(item: ModuleItem) {
```

Comment:
Ich finde die Sagas ziemlich schwer zu lesen, kann aber auch gut sein weil ich Redux Saga nicht gscheid kenn. Das Tool ist wohl schon lange uncool, was ich so gesehen habe managen alle normal den State direkt in der Komponente. Wir könnten eine generische PaginagetItemList machen, die wir kann für jeden Use Case extenden, dann mischen wir nicht immer so komisch die zwei Fälle sondern hätten eine sauberer Kompontente in der alle Logik zusammen ist. Was denkst du?